### PR TITLE
Kevin/addtests

### PR DESCRIPTION
**Pull request by kevindu7**
Adding new tests for miengine

### CODE REVIEW

**Comment by faxue-msft** [comment]
Please squash the commits into one.

File: test/Android/BreakPointWhenTrueCondition/TestScript.xml [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+  </Command>
+  <Command name="eval i">
+    <EvalResult>2</EvalResult>
+    <EvalType>int</EvalType>
+    <EvalName>i</EvalName>
+    <EvalFullName>i</EvalFullName>
+    <EvalIsExpandable>False</EvalIsExpandable>
+  </Command>
+  <Command name="delbps" />
+  <Command name="bp Source1.cpp 20 when_true i&lt;=3" />
```
Comment (faxue-msft) [inline test/Android/BreakPointWhenTrueCondition/TestScript.xml]: Why delete and add the same bp again?

File: test/Android/BreakPointWhenTrueCondition/TestScript.xml [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+  <Command name="bp Source1.cpp 29" />
+  <Command name="go">
+    <Event name="IDebugBreakpointEvent2" expected="True">
+      <functionName>Func</functionName>
+      <function>Func()</function>
+      <file>Source1.cpp</file>
+      <line>29</line>
+      <col>0</col>
+    </Event>
+  </Command>
```
Comment (faxue-msft) [inline test/Android/BreakPointWhenTrueCondition/TestScript.xml]: There isn't that much value repeating similar test for many times

File: test/Android/BreakPointWhenTrueCondition/TestScript.xml [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+  <Command name="bp Source1.cpp 9 when_true i&gt;10" />
+  <Command name="go">
+    <Event name="IDebugBreakpointEvent2" expected="True">
+      <functionName>Func</functionName>
+      <function>Func()</function>
+      <file>Source1.cpp</file>
+      <line>31</line>
+      <col>0</col>
+    </Event>
+  </Command>
```
Comment (faxue-msft) [inline test/Android/BreakPointWhenTrueCondition/TestScript.xml]: too many repeating, instead you might want to test other data types and arithmetic expressions etc.

File: test/Android/BreakPointWhenTrueCondition/TestScript.xml [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+    <Event name="IDebugThreadCreateEvent2" expected="False" />
+    <Event name="IDebugBreakpointEvent2" expected="True">
+      <functionName>Func1</functionName>
+      <function>Func1()</function>
+      <file>Source1.cpp</file>
+      <line>20</line>
+      <col>0</col>
+    </Event>
+  </Command>
+  <!--Verify when true condition in a function-->
```
Comment (edumunoz) [inline test/Android/BreakPointWhenTrueCondition/TestScript.xml]: Sorry I'm not very familiar with glass tests, but what does this comment mean?

**Comment by edumunoz** [comment]
Consider setting the email address associated with your commits to the email address you used to register on GitHub so that the commits show as associated to your GitHub account.

File: test/Android/BreakPointWhenTrueCondition/TestScript.xml (hunk 1 above)
Comment (kevindu7) [inline test/Android/BreakPointWhenTrueCondition/TestScript.xml]: Yes, there is no need to delete the bp. Removed all the "delbps" command.

**Pull request merged**